frdm_k82f:samples/drivers/spi_fujitsu_fram/sample.drivers.spi.fujitsu_fram fails

`sanitycheck -p frdm_k82f -T samples/drivers/spi_fujitsu_fram` fails on master, blocking successful shippable in #21776.  

```
/mnt/nordic/zp/zephyr/samples/drivers/spi_fujitsu_fram/src/main.c: In function 'main':
/mnt/nordic/zp/zephyr/samples/drivers/spi_fujitsu_fram/src/main.c:146:27: error: 'DT_SPI_1_NAME' undeclared (first use in this function); did you mean 'DT_SIM_NAME'?
  spi = device_get_binding(DT_SPI_1_NAME);
                           ^~~~~~~~~~~~~
                           DT_SIM_NAME
```

It appears a legacy devicetree override is being used here that is not defined for this platform.  It's not clear where to add it or what its value should be.

Alternatively this could be switched to use `DT_ALIAS_SPI_1_LABEL`, at the risk of breaking some platform that doesn't provide that as a label.

Alternatively this sample could be converted to devicetree so the proper bus could located automatically.

Alternatively this sample could be removed, since it isn't demonstrating a driver but a hand-coded API for a single chip.

Alternatively....

--- a/samples/drivers/spi_fujitsu_fram/sample.yaml
+++ b/samples/drivers/spi_fujitsu_fram/sample.yaml
@@ -5,4 +5,4 @@ tests:
     tags: drivers fram
     harness: fram
     depends_on: gpio spi
-    filter: CONFIG_SPI_1
+    filter: dt_alias_exists("spi-1")

--- a/samples/drivers/spi_fujitsu_fram/src/main.c
+++ b/samples/drivers/spi_fujitsu_fram/src/main.c
@@ -143,7 +143,7 @@ void main(void)
 
 	printk("fujitsu FRAM example application\n");
 
-	spi = device_get_binding(DT_SPI_1_NAME);
+	spi = device_get_binding(DT_ALIAS_SPI_1_LABEL);
 	if (!spi) {
 		printk("Could not find SPI driver\n");
 		return;

